### DISCUSSION TITLE
Newbie question with modal, how can i refresh the component inside

### DISCUSSION THREAD

**Discussion by narutodiablo** [livewire/livewire#7924] [1. Help]
newbie here, i cannot seem to get this to work. i have a cart component that i want refreshed each time the modal is accessed.

`
<div class="modal" id="test-modal">
     <div class="modal-body">
              @livewire('frontend.cart')
     </div>
</div>

<script>
    document.addEventListener('DOMContentLoaded', function () {
        var modalElement = document.getElementById('test-modal');
        
        modalElement .addEventListener('show.bs.modal', function () {
            Livewire.dispatch('refresh');
        });
    });
</script>

`

in my frontend/cart component code i tried creating a refresh method that copies the content of the mount() method but its not properly refreshing. can someone please help a newbie on how to properly do this. thank you so much

**Comment by danie-ramdhani** [comment]
below your modal element:

```blade
@script
    <script>
        document.getElementById('test-modal').addEventListener('show.bs.modal', function() {
            $wire.$refresh();
        });
    </script>
@endscript
```

don't use `DOMContentLoaded` if you're using livewire navigate feature
